File: src/main/java/com/technews/aggregate/index/HomeController.java
package com.technews.aggregate.index;

import com.technews.aggregate.posts.java.constant.JavaBlogsSubject;
import com.technews.aggregate.posts.spring.constant.SpringBlogsSubject;
import com.technews.aggregate.releases.constant.Categories;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/")
@RequiredArgsConstructor
public class HomeController {

    @GetMapping
    public String home(Model model) {
        model.addAttribute("categories", Categories.values());
        return "index";
    }

    @GetMapping("/blog/spring")
    public String springPosts(Model model) {
        model.addAttribute("categories", SpringBlogsSubject.values());
        return "blog/spring";
    }

    @GetMapping("/blog/java")
    public String JavaPosts(Model model) {
        model.addAttribute("categories", JavaBlogsSubject.values());
        return "blog/java";
    }
}

File: src/main/kotlin/com/technews/aggregate/index/HomeController.kt
package com.technews.aggregate.index

import com.technews.aggregate.releases.constant.Categories
import com.technews.common.constant.JavaBlogsSubject
import com.technews.common.constant.SpringBlogsSubject
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.ui.set
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/")
class HomeController {
    @GetMapping
    fun home(model: Model): String {
        model["categories"] = Categories.entries.toTypedArray()
        return "index"
    }

    @GetMapping("/blog/spring")
    fun springPosts(model: Model): String {
        model["categories"] = SpringBlogsSubject.entries.toTypedArray()
        return "blog/spring"
    }

    @GetMapping("/blog/java")
    fun javaPosts(model: Model): String {
        model["categories"] = JavaBlogsSubject.entries.toTypedArray()
        return "blog/java"
    }
}
